sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuJsonToStructs.scala

```diff
@@ -172,15 +173,23 @@ case class GpuJsonToStructs(
         // Step 1: verify and preprocess the data to clean it up and normalize a few things
         // Step 2: Concat the data into a single buffer
         val (isNullOrEmpty, combined) = cleanAndConcat(input.getBase)
+
         withResource(isNullOrEmpty) { isNullOrEmpty =>
           // Step 3: setup a datasource
           val table = withResource(new JsonDeviceDataSource(combined)) { ds =>
             // Step 4: Have cudf parse the JSON data
-            cudf.Table.readJSON(cudfSchema, jsonOptions, ds)
+            try {
+              cudf.Table.readJSON(cudfSchema, jsonOptions, ds)
+            }catch{
+              case e @ (_: IllegalStateException | _: CudfException) =>
+                logError("Rapids currently does not support all JSON to struct cases of " +
```

Comment:
logging an error before throwing an exception is an anti-pattern.  Whoever catches the exception should be able to take the exception and recover from it. But logging that error can be confusing because there is an error in the logs and no failure.  In this case Spark is going to catch the exception, log it on the executor and then return the exception to the driver to inform the user of the error.  As such it is very unlikely that the end user will see the message being logged here unless they are running in local mode.

If you want to add more information that the end user will see please put the message in the new exception that is wrapping the original ones (instead of "Failed to parse JSON data using cuDF")

Reply:
A new JsonParsingException(subclass of RuntimeExpcetion) has been created. End user will see the info indicating we are not supporting corner cases in JsonParsingException wrapping the original ones.